scripts/SANS/sans/algorithm_detail/scale_helpers.py

```diff
@@ -0,0 +1,161 @@
+from __future__ import (absolute_import, division, print_function)
+import math
+from abc import (ABCMeta, abstractmethod)
+from six import (with_metaclass)
+from sans.common.enums import (SANSInstrument, SampleShape)
+from sans.common.general_functions import create_unmanaged_algorithm
+from sans.common.constants import EMPTY_NAME
+
+
+class DivideByVolume(with_metaclass(ABCMeta, object)):
+    def __init__(self):
+        super(DivideByVolume, self).__init__()
+
+    @abstractmethod
+    def divide_by_volume(self, workspace, scale_info):
+        pass
+
+
+class NullDivideByVolume(DivideByVolume):
+    def __init__(self):
+        super(NullDivideByVolume, self).__init__()
+
+    def divide_by_volume(self, workspace, scale_info):
+        _ = scale_info  # noqa
+        return workspace
+
+
+class DivideByVolumeISIS(DivideByVolume):
+
+    def __init__(self):
+        super(DivideByVolumeISIS, self).__init__()
+
+    def divide_by_volume(self, workspace, scale_info):
+        volume = self._get_volume(scale_info)
+
+        single_valued_name = "CreateSingleValuedWorkspace"
+        single_valued_options = {"OutputWorkspace": EMPTY_NAME,
+                                 "DataValue": volume}
+        single_valued_alg = create_unmanaged_algorithm(single_valued_name, **single_valued_options)
+        single_valued_alg.execute()
+        single_valued_workspace = single_valued_alg.getProperty("OutputWorkspace").value
+
+        divide_name = "Divide"
+        divide_options = {"LHSWorkspace": workspace,
+                          "RHSWorkspace": single_valued_workspace}
+        divide_alg = create_unmanaged_algorithm(divide_name, **divide_options)
+        divide_alg.setPropertyValue("OutputWorkspace", EMPTY_NAME)
+        divide_alg.setProperty("OutputWorkspace", workspace)
+        divide_alg.execute()
+        return divide_alg.getProperty("OutputWorkspace").value
+
+    def _get_volume(self, scale_info):
```

Comment:
Just a side note: there is now a `volume()` method in Mantid's shape `Object` which can be used to calculate the volume of an arbitrary shape. I was planning to use this in conjunction with the `SetSample` algorithm handle the absolute normalization in our time-of-flight workflow.